### ISSUE TITLE
Column UserAgent in Visitor table is too short, causes exception with some agents

### ISSUE THREAD

**Issue by chlupac** [oqtane/oqtane.framework#2337]
Size of Visitor.UserAgent should be increased. 
I would make the change, but I don't know exactly how to add migration in Oqtane.Server.

```
Microsoft.EntityFrameworkCore.DbUpdateException: An error occurred while saving the entity changes. See the inner exception for details.
 ---> Microsoft.Data.SqlClient.SqlException (0x80131904): String or binary data in table mfl.dbo.Visitor, column UserAgent would be shortened. Shortened value: Mozilla/5.0 (iPhone; CPU iPhone OS 15_5 like Mac OS X) AppleWebKit/605.1.15 (KHTML, like Gecko) Mobi.
Příkaz byl ukončen.
   at Microsoft.Data.SqlClient.SqlConnection.OnError(SqlException exception, Boolean breakConnection, Action`1 wrapCloseInAction)
   at Microsoft.Data.SqlClient.SqlInternalConnection.OnError(SqlException exception, Boolean breakConnection, Action`1 wrapCloseInAction)
   at Microsoft.Data.SqlClient.TdsParser.ThrowExceptionAndWarning(TdsParserStateObject stateObj, Boolean callerHasConnectionLock, Boolean asyncClose)
   at Microsoft.Data.SqlClient.TdsParser.TryRun(RunBehavior runBehavior, SqlCommand cmdHandler, SqlDataReader dataStream, BulkCopySimpleResultSet bulkCopyHandler, TdsParserStateObject stateObj, Boolean& dataReady)
   at Microsoft.Data.SqlClient.SqlDataReader.TryConsumeMetaData()
   at Microsoft.Data.SqlClient.SqlDataReader.get_MetaData()
   at Microsoft.Data.SqlClient.SqlCommand.FinishExecuteReader(SqlDataReader ds, RunBehavior runBehavior, String resetOptionsString, Boolean isInternal, Boolean forDescribeParameterEncryption, Boolean shouldCacheForAlwaysEncrypted)
   at Microsoft.Data.SqlClient.SqlCommand.RunExecuteReaderTds(CommandBehavior cmdBehavior, RunBehavior runBehavior, Boolean returnStream, Boolean isAsync, Int32 timeout, Task& task, Boolean asyncWrite, Boolean inRetry, SqlDataReader ds, Boolean describeParameterEncryptionRequest)
   at Microsoft.Data.SqlClient.SqlCommand.RunExecuteReader(CommandBehavior cmdBehavior, RunBehavior runBehavior, Boolean returnStream, TaskCompletionSource`1 completion, Int32 timeout, Task& task, Boolean& usedCache, Boolean asyncWrite, Boolean inRetry, String method)
   at Microsoft.Data.SqlClient.SqlCommand.ExecuteReader(CommandBehavior behavior)
   at Microsoft.Data.SqlClient.SqlCommand.ExecuteDbDataReader(CommandBehavior behavior)
   at Microsoft.EntityFrameworkCore.Storage.RelationalCommand.ExecuteReader(RelationalCommandParameterObject parameterObject)
   at Microsoft.EntityFrameworkCore.Update.ReaderModificationCommandBatch.Execute(IRelationalConnection connection)
ClientConnectionId:4a4d1f7f-6cf0-4214-a0d0-db797f97d1c9
Error Number:2628,State:1,Class:16
   --- End of inner exception stack trace ---
   at Microsoft.EntityFrameworkCore.Update.ReaderModificationCommandBatch.Execute(IRelationalConnection connection)
   at Microsoft.EntityFrameworkCore.Update.Internal.BatchExecutor.Execute(IEnumerable`1 commandBatches, IRelationalConnection connection)
   at Microsoft.EntityFrameworkCore.ChangeTracking.Internal.StateManager.SaveChanges(IList`1 entriesToSave)
   at Microsoft.EntityFrameworkCore.ChangeTracking.Internal.StateManager.SaveChanges(StateManager stateManager, Boolean acceptAllChangesOnSuccess)
   at Microsoft.EntityFrameworkCore.SqlServer.Storage.Internal.SqlServerExecutionStrategy.Execute[TState,TResult](TState state, Func`3 operation, Func`3 verifySucceeded)
   at Microsoft.EntityFrameworkCore.ChangeTracking.Internal.StateManager.SaveChanges(Boolean acceptAllChangesOnSuccess)
   at Microsoft.EntityFrameworkCore.DbContext.SaveChanges(Boolean acceptAllChangesOnSuccess)
   at Oqtane.Repository.VisitorRepository.AddVisitor(Visitor visitor) in /Users/pavel/projects/lib/oqtane.framework/Oqtane.Server/Repository/VisitorRepository.cs:line 28
   at Oqtane.Pages.HostModel.TrackVisitor(Int32 SiteId) in /Users/pavel/projects/lib/oqtane.framework/Oqtane.Server/Pages/_Host.cshtml.cs:line 313
```

**Comment by sbwalker** [comment]
User Agent strings are passed in the header and there is no maximum physical size constraint on headers. For practical purposes the general recommendation is that User Agents should not exceed 256 characters (https://developers.whatismybrowser.com/learn/browser-detection/user-agents/user-agent-best-practices). Oqtane currently defines the max length of the UserAgent column as 256 characters (at least for those databases which support it as not all relational databases support a max string length specification).  

So there are a few different options here to consider:

1. Simply truncate the User Agent string at 256 characters before saving it in the database. Generally speaking User Agent strings which are greater than 256 characters have been customized and often carry malicious payloads. 
2. Expand the size of the UserAgent column in the database
3. Do a combination of 1 & 2

My preference is the first option. Your thoughts?

**Comment by chlupac** [comment]
I agree with first option. Simple.

**Comment by sbwalker** [comment]
resolved by #2338
